python/tvm/contrib/target/onnx.py

```diff
@@ -0,0 +1,752 @@
+           otherwise by default no attributes are passed
+        """
+        return {}
+
+    @classmethod
+    def convert(cls, node, model_container, node_list):
+        attrs = cls.convert_attributes(node['node'].attrs)
+        node = onnx.helper.make_node(cls.__name__,
+                                     node['input_names'],
+                                     node['output_names'],
+                                     **attrs)
+        model_container.add_nodes([node])
+
+
+def rename(op_name):
+    """ This method creates dynamic operator of name op_name with empty attributes
+    """
+    return type(op_name, (OpConverter,), {})
+
+
+class Reshape(object):
+    """ Operator converter for Reshape.
+    """
+
+    @classmethod
+    def convert(cls, node, model_container, node_list):
+        """Converts Relay operator Reshape to ONNX operator.
+           Relay operator accepts shape as attribute but ONNX operator
+           accepts it as a input.
+        """
+
+        shape = numpy.asarray([a.value for a in node['node'].attrs.newshape],
+                              dtype=numpy.int64)
+        input_name = 'shape{}'.format(node['output_names'][0])
+        node = onnx.helper.make_node(cls.__name__, [node['input_names'][0], input_name],
+                                     node['output_names'])
+        model_container.add_nodes([node])
+        add_input(shape, input_name, model_container)
+
+
+class Conv(OpConverter):
+    """ Operator converter for Conv.
+    """
+
+    @classmethod
+    def convert_attributes(cls, attrs):
+        return {
+            'group': attrs.get_int("groups"),
+            'pads': attrs.get_int_tuple("padding"),
+            'strides': attrs.get_int_tuple("strides"),
+            'dilations': attrs.get_int_tuple("dilation"),
+            'kernel_shape': attrs.get_int_tuple("kernel_size"),
+        }
+
+
+class MaxPool(OpConverter):
+    """ Operator converter for MaxPool.
+    """
+
+    @classmethod
+    def convert_attributes(cls, attrs):
+        return {
+            'pads': attrs.get_int_tuple("padding"),
+            'strides': attrs.get_int_tuple("strides"),
+            'kernel_shape': attrs.get_int_tuple("pool_size"),
+        }
+
+
+class Transpose(OpConverter):
+    """ Operator converter for Transpose.
+    """
+
+    @classmethod
+    def convert_attributes(cls, attrs):
+        return {'perm': attrs.get_int_tuple("axes")} if attrs["axes"] else {}
+
+
+class MatMul(OpConverter):
+    """ Operator converter for MatMul.
+    """
+
+    @classmethod
+    def convert(cls, node, model_container, node_list):
+        output_name = 'inter{}'.format(node['output_names'][0])
+        transpose_node = onnx.helper.make_node(Transpose.__name__,
+                                               [node['input_names'][1]],
+                                               [output_name],
+                                               **{'perm': (1, 0)})
```

Comment:
Isn't `perm=(1, 0)` simpler?

Reply:
Yes it is. Corrected at other places as well.